src/main/java/org/skriptlang/skript/bukkit/itemcomponents/blocking/elements/ExprBlockCompReductions.java

```diff
@@ -0,0 +1,120 @@
+package org.skriptlang.skript.bukkit.itemcomponents.blocking.elements;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import io.papermc.paper.datacomponent.item.blocksattacks.DamageReduction;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.itemcomponents.blocking.BlockingExperimentalSyntax;
+import org.skriptlang.skript.bukkit.itemcomponents.blocking.BlockingWrapper;
+import org.skriptlang.skript.bukkit.itemcomponents.blocking.DamageReductionWrapper;
+import org.skriptlang.skript.registration.SyntaxRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("Blocking Component - Damage Reductions")
+@Description("""
+	The damage reductions applied to the item.
+	Damage Reductions contain data that attribute to:
+		- What damage types can be being blocked
+		- The base amount of damage to block when blocking one of the damage types
+		- The factor amount of damage to block when blocking one of the damage types
+		- The angle at which the item can block when blocking one of the damage types
+	NOTE: Blocking component elements are experimental. Thus, they are subject to change and may not work as intended.
+	""")
+@Example("set {_reductions::*} to the damage reductions of {_item}")
+@Example("""
+	set {_reduction} to a custom damage reduction
+	add {_reduction} to the damage reductions of {_item}
+	""")
+@RequiredPlugins("Minecraft 1.21.5+")
+@Since("INSERT VERSION")
+public class ExprBlockCompReductions extends PropertyExpression<BlockingWrapper, DamageReductionWrapper> implements BlockingExperimentalSyntax {
```

Comment:
like we discussed with item damage functions, can we add some sugar for making these?
```
a damage reduction that blocks %number% heart[s] and %number%\% of %damage types% damage
a damage reduction that blocks 1 heart and 50% of magic damage
```